### PR TITLE
derive attachment fname from BufferedReader object (#1437) 

### PR DESCRIPTION
**Pull request by mtripoli**
Offering a fix for #1437.

### CODE REVIEW

**Review by adehad — CHANGES_REQUESTED** [review]
Thanks @mtripoli, change looks good to me I think
test failure isn't related to this, but could I ask if you can add a test case here:
https://github.com/pycontribs/jira/blob/main/tests/resources/test_attachment.py

Can be a very simple one we we just include the first argument and if no exception is raised and the attachment is found we are all good. Hopefully the existing cases are sufficient to help you crack on, if not let me know and I can write up a test case

**Comment by mtripoli** [comment]
@adehad , done. I totally missed the tests/resources dir otherwise I would have provided the coverage for the change on the first go.

File: tests/resources/test_attachment.py [hunk 1]
```diff
@@ -30,6 +31,14 @@ def test_1_add_remove_attachment_using_filestream(self):
...
     def test_2_add_remove_attachment_using_filename(self):
         issue = self.jira.issue(self.issue_1)
+        attachment_no_filename_specified = self.jira.add_attachment(
+            issue,
+            TEST_ATTACH_PATH,
+        )
+        new_attachment = self.jira.attachment(attachment_no_filename_specified.id)
+        msg = f"attachment, no filename specified {new_attachment.__dict__} of issue {issue}"
+        self.assertEqual(new_attachment.filename, Path(TEST_ATTACH_PATH).name, msg=msg)
+        #
```
Comment (adehad) [inline tests/resources/test_attachment.py:41]: excellent test case, however, let's put this into a new method, the test name doesn't make sense here

We don't need to add a number to the tests so just something like:


```suggestion
    def test_add_attachment_without_filename(self):
        issue = self.jira.issue(self.issue_1)
        attachment_no_filename_specified = self.jira.add_attachment(
            issue=issue,
            attachment=TEST_ATTACH_PATH,
            filename=None,
        )
        new_attachment = self.jira.attachment(attachment_no_filename_specified.id)
        msg = f"attachment, no filename specified {new_attachment.__dict__} of issue {issue}"
        assert new_attachment.filename == Path(TEST_ATTACH_PATH).name, msg

    def test_2_add_remove_attachment_using_filename(self):
        issue = self.jira.issue(self.issue_1)
```

(I've also moved away from the self.assert methods, one day we will clean out the legacy unittest syntax)

**Review by adehad — CHANGES_REQUESTED** [review]

**Comment by adehad** [comment]
> @adehad , done. I totally missed the tests/resources dir otherwise I would have provided the coverage for the change on the first go.

No problem at all, the tests are definitely hard to understand for new contributors, its on my list to update the contributing guide to clarify things

**Review by adehad — APPROVED** [review]
Awesome, thanks for this !

**Pull request merged**